scrapy/core/downloader/handlers/http11.py

```diff
@@ -91,11 +92,15 @@ def __init__(self, reactor, host, port, proxyConf, contextFactory,
 
     def requestTunnel(self, protocol):
         """Asks the proxy to open a tunnel."""
-        tunnelReq = 'CONNECT %s:%s HTTP/1.1\r\n' % (self._tunneledHost,
-                                                  self._tunneledPort)
+        tunnelReq = (
+            b'CONNECT ' +
+            to_bytes(self._tunneledHost, encoding='ascii') + b':' +
+            to_bytes(str(self._tunneledPort)) +
+            b' HTTP/1.1\r\n')
```

Comment:
yea, was just thinking that -- would be cleaner.
the string concatenation reminds of Java too much. :)